From 9ae06115b3cf2c2089d803cf78d1964c21d0998b Mon Sep 17 00:00:00 2001
From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
Date: Mon, 29 Sep 2025 11:23:04 +0800
Subject: [PATCH 1/2] sdhci-efx: enable cache flush for rv32

Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
---
 drivers/mmc/host/sdhci-efx.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci-efx.c b/drivers/mmc/host/sdhci-efx.c
index ebb9cb779e64..0c983473f505 100644
--- a/drivers/mmc/host/sdhci-efx.c
+++ b/drivers/mmc/host/sdhci-efx.c
@@ -41,13 +41,15 @@ static u32 efx_readl(struct sdhci_host *host, int reg)
 {
 	u32 val;
 	u32 offset = reg + EFX_SDHCI_REG_OFFSET;
+#if defined(CONFIG_32BIT)
 	struct mmc_command *data_cmd = host->data_cmd;
 
-	// clear cache for ACMD51
+	// clear cache for ACMD51. The clear cache only works for Efinix RV32 SoC.
 	if (data_cmd) {
                 if (data_cmd->opcode == 51)
                         asm(".word(0x500F)");
         }
+#endif /* CONFIG_32BIT */
 
 	val = readl(host->ioaddr + offset);
 
-- 
2.43.0

