From 5851856323565b70c997df009b3a26b507bae98a Mon Sep 17 00:00:00 2001
From: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
Date: Mon, 29 Sep 2025 11:23:33 +0800
Subject: [PATCH 2/2] efinix_tse: enable cache flush for rv32

Signed-off-by: Mohamad Noor Alim Hussin <mnalim@efinixinc.com>
---
 drivers/net/ethernet/efinix/efinix_tse.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/efinix/efinix_tse.h b/drivers/net/ethernet/efinix/efinix_tse.h
index f27bd8266719..a666bcb743c6 100644
--- a/drivers/net/ethernet/efinix/efinix_tse.h
+++ b/drivers/net/ethernet/efinix/efinix_tse.h
@@ -253,6 +253,7 @@ asm(".set regnum_t4  , 29");
 asm(".set regnum_t5  , 30");
 asm(".set regnum_t6  , 31");
 
+#if defined(CONFIG_32BIT)
 //Invalidate the whole data cache
 #define data_cache_invalidate_all() asm(".word(0x500F)");
 
@@ -265,6 +266,10 @@ asm(".set regnum_t6  , 31");
      : "r" (address)                               \
     );                                         \
 })
+#else
+#define data_cache_invalidate_all()
+#define data_cache_invalidate_address(address)
+#endif /* CONFIG_32_BIT */
 
 //Invalidate the whole instruction cache
 #define instruction_cache_invalidate() asm("fence.i");
-- 
2.43.0

